src/workerd/util/duration-alert.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <kj/string.h>
+#include <kj/time.h>
+#include <kj/debug.h>
+
+namespace workerd::util {
+
+class DurationAlert {
+// This is a utility class for instantiating a timer, which will log if it's destructed after a specified time
+// This works by relying on RAII (Scope-Bound Resource Management) to check how much time has elapsed
+// when the object is destructed. This ensures that the time is checked when the timer object goes out of scope
+// thereby timing everything after the initialization of the timer, until the end of the scope where it was
+// instantiated.
+public:
+  DurationAlert(const kj::MonotonicClock& clock, kj::Duration logWarningTime, kj::StringPtr logMessage)
+    : logWarningTime(logWarningTime),
+      logMessage(logMessage),
+      start(clock.now()),
+      clock(clock) {}
+
+  ~DurationAlert() noexcept(false) {
+    kj::Duration duration = clock.now() - start;
+    if (duration >= logWarningTime) {
+      KJ_LOG(WARNING, "NOSENTRY", logMessage, duration);
```

Comment:
```suggestion
      KJ_LOG(WARNING, kj::str("NOSENTRY ", logMessage), duration);
```

Reply:
Actually, we already have a macro for handling the `NOSENTRY` bit here, don't we?

Reply:
Heh... per earlier discussion, the current implementation of that macro requires a literal string as the first argument, so you'd have to do something like this anyway, to get it to work with the logMessage member variable.

https://github.com/cloudflare/workerd/pull/1693#discussion_r1495678123